autometacal/python/datasets/galaxies.py

```diff
@@ -0,0 +1,173 @@
+import galsim
+import tensorflow as tf
+import numpy as np
+from scipy.stats import truncnorm, norm
+
+"""
+Simple wrappers for toy galaxy generation
+Functions with gs_ prefix depend on galsim
+"""
+
+
+def gs_generate_images(**kwargs):
+  
+  """ Random Galaxy Generator
+  Generates noiseless galaxy images with a simple light profile. The resulting image is before the convolution with a PSF. 
+  Galaxy shapes follow a bivariate normal distribution centered in zero.
+  
+  Args:
+    g_range: galaxy shapes go from -g_range and + g_range in each g1, g2 
+    g_scatter: galaxy shapes scatter
+    flux: galaxy flux (counts)
+    pixel_scale: intended pixel scale in arcsec^2/pixel
+    stamp_size: size in pixels of the NxN resulting image
+    method: galsim drawing method
+    interpolator: galsim interpolation used to draw the image on the pixel grid.
+  Returns:
+    g1, g2: galaxy shape parameters
+    gal_image.array: numpy array that represents galaxy image
+  
+  """
+
+  defaults = {'g_range' : 0.8,        #elipticity
+              'g_scatter' : 0.25,     #
+              'mean_radius': 3.0,     #size
+              'scatter_radius': 0.1,  #
+              'psf_beta': 5,          #psf
+              'psf_fwhm': 0.5,        #
+              'mean_snr': 200,        #snr
+              'scatter_snr': 20,      #
+              'flux' : 1e5,           #flux
+              'pixel_scale' : 0.2,    #
+              'stamp_size' : 50,      #
+              'method' : "no_pixel"   #
+             }
+
+  defaults.update(kwargs)
+  
+  #ellipticity range
+  a, b = (-defaults['g_range'] - 0) / defaults['g_scatter'], (defaults['g_range'] - 0) / defaults['g_scatter']
+  g1=g2=1
+  
+  #select ellipticities, ensure g<=1
+  while g1**2+g2**2>1:
+    g1 = truncnorm.rvs(a, b, loc=0, scale=defaults['g_scatter'])
+    g2 = truncnorm.rvs(a, b, loc=0, scale=defaults['g_scatter'])
+    
+  re = norm.rvs(3, 0.1)
+  
+  #very simple galaxy model
+  gal = galsim.Exponential(flux=defaults['flux'] ,
+                           half_light_radius=re)
+  
+  #apply shear
+  gal = gal.shear(g1=g1,g2=g2)
+    
+  
+  #create constant psf
+  psf = galsim.Moffat(beta=defaults['psf_beta'], 
+                      fwhm=defaults['psf_fwhm'])
+  
+  #draw galaxy before convolution
+  true_gal_image = gal.drawImage(nx=defaults['stamp_size'],
```

Comment:
is this ever used?

Reply:
Not atm, but it was an idea of having the noiseless image separated if we wanted to change the noise.